pkg/controller/apicast/apicast_controller.go

```diff
@@ -189,6 +188,34 @@ func (r *ReconcileAPIcast) Reconcile(request reconcile.Request) (reconcile.Resul
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileAPIcast) updateStatus(instance *appsv1alpha1.APIcast, reconciler *APIcastLogicReconciler) (reconcile.Result, error) {
+	apicast, err := reconciler.APIcastFromCRContents()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	var apicastDeployment *appsv1.Deployment
+	r.Client().Get(context.TODO(), types.NamespacedName{Name: apicast.DeploymentName, Namespace: apicast.Namespace}, apicastDeployment)
+	if err != nil && !errors.IsNotFound(err) {
+		return reconcile.Result{}, err
+	}
+
+	if err != nil && errors.IsNotFound(err) {
+		return reconcile.Result{}, err
+	}
+
+	deployedImage := apicastDeployment.Spec.Template.Spec.Containers[0].Image
+	if !reflect.DeepEqual(instance.Status.Image, deployedImage) {
```

Comment:
Compare two strings with `reflect.DeepEqual` is like "kiling flies with cannons"?

Reply:
It can be changed to simple comparison yes.